src/clib/Makefile

```diff
@@ -72,6 +72,13 @@ include $(MAKE_CONFIG_OVERRIDE)
 -include $(GRACKLE_DIR)/Make.mach.$(CONFIG_MACHINE)
 -include $(HOME)/.grackle/Make.mach.$(CONFIG_MACHINE)
 
+#-----------------------------------------------------------------------
+# OTHER_DEFINES is used to specify a macro indicating that legacy
+# functions defined inline in the main header file should be omitted
+# while compiling the library
+#-----------------------------------------------------------------------
+OTHER_DEFINES = -DOMIT_LEGACY_INTERNAL_GRACKLE_FUNC
```

Comment:
Unless I'm misunderstanding, I believe this means that the legacy functions are not compiled by default. If that's the case, I think my preference would be the reverse, i.e., to make legacy compilation opt-out rather than opt-in. My reasoning is that this will give users a heads-up when they compile Grackle for their simulation codes instead of legacy functions simply being unavailable by default when they go to link with their simulation code. I'm open to other opinions, though.

Reply:
Oh right, this is probably not well-enough explained. In short, the default behavior is equivalent to what you want: the user will have access to the legacy functions in their own simulation-code unless they explicitly define the ``OMIT_LEGACY_INTERNAL_GRACKLE_FUNC``  macro while compiling their simulation code.

### In more detail:

- as you're aware: the  the ``_initialize_chemistry_data``, ``_set_default_chemistry_parameters``, and ``_free_chemistry_data`` legacy functions are conditionally all declared/defined (with the ``inline`` specifier) in ``grackle.h`` when the ``OMIT_LEGACY_INTERNAL_GRACKLE_FUNC`` macro is not defined
- This means that these functions definitions are included in **every** object file that is produced from a source file that includes ``grackle.h`` and does not define ``OMIT_LEGACY_INTERNAL_GRACKLE_FUNC``. This applies to the object files produced as part of compiling of the Grackle library **AND** object files compiled as part of compiling a user's simulation library.

I had modified the Makefile to define the ``OMIT_LEGACY_INTERNAL_GRACKLE_FUNC`` macro while compiling source files for the Grackle library since we don't need the function definitions within the Grackle library. However, the definition of this macro has no effect on external applications that link against the Grackle library.

In other words, all 3 of these legacy functions will be defined in a user's simulation code in any files where they include ``"grackle.h"`` (unless they explicitly define the ``OMIT_LEGACY_INTERNAL_GRACKLE_FUNC`` macro).

Let me know if this is confusing - I can try to rephrase this.

### How to resolve this:

Going forward, I see two main options to try to avoid this confusion:
1. The simplest approach is to just drop this idea of conditional compilation. The deprecation notices are probably adequate to warn users that they should stop using these functions.
2. If my explanation here makes sense, I could use it to update the description within the Makefile right before we define the flag.

Let me know what you preference is (I'm thinking that simplicity might be the better option here).

Reply:
Ah, I get it now. `OMIT_LEGACY_INTERNAL_GRACKLE_FUNC` is being defined for the compile of the Grackle library itself. Users are to add this to the compile of their simulation codes if they don't want those functions to be present. I don't think you need to change anything. This should be clear when users compile their simulation codes.